### PR TITLE
chore: Updates project resource with new SDK

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates project resource with new SDK.

- Teams and Limits can be updated to empty.
- Checked read-only fields: 
  - Group: Id,  ClusterCount (not used), Created (not used), Links (not used), 
  - TeamRole: Links (not used)
  - DataFederationLimit: CurrentUsage, DefaultLimit, MaximumLimit, Name, Value, LastModifiedDate (not used)
- Checked write-only fields:
  - Group: WithDefaultAlertsSettings

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-222714

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: internal/service/project/resource_project_test.go [hunk 1]
```diff
@@ -543,22 +543,11 @@ func TestAccProjectRSProject_basic(t *testing.T) {
...
-				Config: acc.ConfigProject(projectName, orgID, []*admin.TeamRole{}),
-				Check: resource.ComposeTestCheckFunc(
-					acc.CheckProjectExists(resourceName, &group),
-					acc.CheckProjectAttributes(&group, projectName),
-					resource.TestCheckResourceAttr(resourceName, "name", projectName),
-					resource.TestCheckResourceAttr(resourceName, "org_id", orgID),
-					resource.TestCheckResourceAttr(resourceName, "cluster_count", clusterCount),
-					resource.TestCheckNoResourceAttr(resourceName, "teams.#"),
-				),
-			},
```
Comment (lantoli) [inline internal/service/project/resource_project_test.go:556]: this test logic is checked in updatedToEmptyTeams

File: internal/service/project/model_project.go [hunk 2]
```diff
@@ -119,11 +119,11 @@ func newTFTeamsResourceModel(ctx context.Context, atlasTeams *admin.PaginatedTea
 
 func NewTeamRoleList(ctx context.Context, teams []TfTeamModel) *[]admin.TeamRole {
 	res := make([]admin.TeamRole, len(teams))
-
 	for i, team := range teams {
+		roleNames := conversion.TypesSetToString(ctx, team.RoleNames)
 		res[i] = admin.TeamRole{
 			TeamId:    team.TeamID.ValueStringPointer(),
-			RoleNames: conversion.NonEmptyToPtr(conversion.TypesSetToString(ctx, team.RoleNames)),
+			RoleNames: &roleNames,
```
Comment (lantoli) [inline internal/service/project/model_project.go:126]: no need to use NonEmptyToPtr, TypesSetToString always returns a non-nil object so we can always take its address

File: internal/service/project/resource_project.go [hunk 3]
```diff
@@ -828,3 +799,17 @@ func getChangesInLimitsSet(planLimits, stateLimits []TfLimitModel) (newElements,
...
+	}
+}
+
+func updateProjectBool(plan, state types.Bool, setting **bool) bool {
+	if plan != state {
+		*setting = plan.ValueBoolPointer()
+		return true
+	}
+	return false
+}
```
Comment (marcosuma) [inline internal/service/project/resource_project.go]: nice you had this idea of moving to separate method. Can we unit test this now?

File: internal/service/project/resource_project.go [hunk 4]
```diff
@@ -29,13 +29,14 @@ import (
 )
 
 const (
-	projectResourceName            = "project"
+	ErrorProjectRead               = "error getting project(%s): %s"
```
Comment (marcosuma) [inline internal/service/project/resource_project.go]:
```suggestion
	ErrorProjectRead               = "error getting project (%s): %s"
```

Comment (lantoli) [reply internal/service/project/resource_project.go]: thx, changed

File: internal/service/project/resource_project.go (hunk 3 above)
Comment (lantoli) [reply internal/service/project/resource_project.go]: done

**Review by marcosuma — APPROVED** [review]

**Pull request merged**